Mathlib/FieldTheory/Finite/Basic.lean

```diff
@@ -533,6 +533,12 @@ theorem pow_card_sub_one_eq_one {p : ℕ} [Fact p.Prime] {a : ZMod p} (ha : a 
     rwa [ZMod.card p] at h
 #align zmod.pow_card_sub_one_eq_one ZMod.pow_card_sub_one_eq_one
 
+lemma pow_card_sub_one {p : ℕ} [Fact p.Prime] (a : ZMod p) :
+    a ^ (p - 1) = if a ≠ 0 then 1 else 0 := by
```

Comment:
Why test on `a ≠ 0` instead of `a = 0`? Does it make sense to use `Pi.mulSingle` here?

Reply:
The point is that `if p then 1 else 0` is used in plenty of statements, eg `Finset.sum_boole`. I'm happy to add other versions later but this is precisely the one I need for now